001-finding_resources.md

```diff
@@ -0,0 +1,120 @@
+4. A SimulationRunner is then instantiated for each `sdf::World`. The
+   LevelManager, contained in each SimulationRunner, loads each `sdf::Model`
+   and creates the appropriate set of components. Plugin loading is conducted
+   during this step, but no mesh URI resource loading is conducted.
+
+5. Systems and other aspects of simulation, such as rendering, must then each
+   resolve URI resources contained within the loaded world and models.
+   The `Util::asFullPath` is a helper function that is used to convert a URI
+   into a system path. This helper function only alters URIs that have
+   a relative path. URIs that have a scheme or are a full path are
+   untouched.
+
+6. The result of `Util::asFullPath` is then passed to
+   `ignition::common::MeshManager::Load`, which then calls
+   `ignition::common::findFile` to finally resolve the location of the URI
+   resource.
+
+7. `ignition::common::findFile` executes the following procedure.
+
+    1. Calls `SystemPaths::FindFile(string file, bool searchLocalPath)`
+    2. `SystemPaths::FindFile(string file, bool searchLocalPath)` executes:
+        1. if `file` is empty, then return an empty string
+        2. if `file` is a valid URI, then `path`
+           = `SystemsPath::FindFileURI(file)`. The `FindFileURI(file)`
+           executes:
+            1. if the URI scheme is "file", then `path` = `SystemsPath::FindFile(_file)`. This will only be executed if `SystemsPath::FindFileURI(file)` is called indepently of `SystemPaths::FindFile`.
+            2. else if there is a user define CB, the `path` = `FileURICB(_file)`. This `FileURICB` is deprecated in ign-common3.
+            4. if `path` is empty, try each optional `FileURICbs`. The
+               vector of `FileURICbs` replaces the deprecated `FileURICB`.
+            5. if `path` is empty or `path` doesn't exist, then return empty string
+            6. return `path`.
+        3. else if `file` is an absolute file path, then `path` = `file`.
+        4. else if `searchLocalPath == true` and the `file` exists in the
+           current working directory, then `path` = current working
+           directory + `file`.
+        5. else if `file` is an absolute path and exists, then `path` = `file`
+        6. else `path` = `findFileCB(file)`. `findFileCb` is an optional user
+           defined callback.
+        7. if `path` is empty, search the set of defined `filePaths`.
+        8. if `path` is empty, call the optional user defined callbacks
+        9. if `path` is empty or `path` does not exist, return an empty string.
+        10. return `path`.
+
+#### Issues with the current process
+
+Problems with the current process exist in the last step, when `ignition::common::findFile` is executed.
+
+The problems are:
+
+1. Any scheme that is not "file", will not resolve unless a user defines
+   a custom callback. The "model" scheme is particularly common. ROS makes
+   use of the "package" scheme, which may sneak into SDF files. There is
```

Comment:
Just a note that the SDF -> URDF parser automatically converts `package://` to `model://`

Reply:
Clarified this in a3696f8e717265fb69c906bc630f83723b50d8dd